dfusion_rust_core/src/util.rs

```diff
@@ -1,12 +1,25 @@
-use std::str::FromStr;
+use std::convert::TryInto;
 use web3::types::U256;
 
+/// Convert a u128 to a U256.
 pub fn u128_to_u256(x: u128) -> U256 {
     U256::from_big_endian(&x.to_be_bytes())
 }
 
+/// Convert a U256 to a u128.
+///
+/// # Panics
+///
+/// Panics if the U256 overflows a u128.
 pub fn u256_to_u128(x: U256) -> u128 {
-    u128::from_str(&x.to_string()).unwrap()
+    let mut bytes = [0u8; 32];
+    x.to_big_endian(&mut bytes[..]);
+
+    let hi = u128::from_be_bytes(bytes[..16].try_into().expect("correct slice length"));
+    let lo = u128::from_be_bytes(bytes[16..].try_into().expect("correct slice length"));
+
+    assert_eq!(hi, 0, "U256 to u128 overflow");
+    lo
```

Comment:
Was there something wrong with relying on the errors that I would assume are emitted upon overflow from `u128::from_str`? We all thought the one liner was more elegant and relied on already tested code.

Reply:
It was a bit of nit-pick of mine to convert an int to a string and then back to an int when really just the LSBs of the larger int need to be read.